Sources/performance-worker/BenchmarkService.swift

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import GRPCCore
+
+import struct Foundation.Data
+
+@available(macOS 13.0, iOS 16.0, watchOS 9.0, tvOS 16.0, *)
+struct BenchmarkService: Grpc_Testing_BenchmarkService.ServiceProtocol {
+  /// One request followed by one response.
+  /// The server returns the client payload as-is.
+  func unaryCall(
+    request: GRPCCore.ServerRequest.Single<Grpc_Testing_BenchmarkService.Method.UnaryCall.Input>
+  ) async throws
+    -> GRPCCore.ServerResponse.Single<Grpc_Testing_BenchmarkService.Method.UnaryCall.Output>
+  {
+
+    // Throw an error if the status is not `ok`. Otherwise, an `ok` status is automatically sent
+    // if the request is successful.
+    if request.message.responseStatus.isInitialized {
+      try self.echoStatus(responseStatus: request.message.responseStatus)
+    }
+
+    return ServerResponse.Single(
+      message: Grpc_Testing_BenchmarkService.Method.UnaryCall.Output.with {
+        $0.payload = request.message.payload
+      }
```

Comment:
This isn't quite right: the request tells us how big the response payload should be (via `responeSize`). Same goes for all the other RPCs.

Reply:
is this also true for `streamingBothWays()`? The documentation for it says "Both sides send the content of their own choice to the other."

Reply:
Yes, I believe so, I think the docs are outdated because the scenario descriptions which drive the tests expect different request/response sizes.

Reply:
Wouldn't making this change mean that `streamingCall()` and `streamingBothWays()` become the same function?